integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/projection/UnsupportedNestingProjectionBaseIT.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.search.projection;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMapperUtils.documentProvider;
+
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.IndexObjectFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaObjectField;
+import org.hibernate.search.engine.backend.types.ObjectStructure;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class UnsupportedNestingProjectionBaseIT {
+
+	@Rule
+	public final SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	private final SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new )
+			.name( "my-index" );
+
+	@Before
+	public void setup() {
+		setupHelper.start().withIndexes( index ).setup();
+		initData();
+	}
+
+	@Test
+	public void id() {
+		assertThatThrownBy( () -> index.createScope().query()
+				.select( f -> f.object( "nested" ).from(
+								f.id()
+						).asList().multi()
+				)
+				.where( f -> f.matchAll() )
+				.toQuery()
+		).isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"'ID' projections are only applicable within the root projection context",
+						"An ID projection represents the document ID and adding it as a part of the nested object projection might produce misleading results since it is always a root document ID and not a nested object ID."
+
+				);
+	}
+
+	@Test
+	@Ignore("entity projection might not load an entity but do a composite projection from the root context instead...")
```

Comment:
> not sure how to go about entity projection. Since it not necessarily reaches the projection, but might instead use a composite projection...

Maybe it would be a good idea to introduce an "EntityCompositeProjection" to model the fact that such a composite projection is supposed to return the root entity?

Basically it would just be a wrapper projection that would delegate to the composite projection, with an additional check in `request` to prevent it from being used in nested contexts... We wouldn't expose it in APIs, just in a new `SearchProjection<T> entityComposite(SearchProjection<T> delegate)`  method in `org.hibernate.search.engine.search.projection.spi.SearchProjectionBuilderFactory`. We'd call it from `org.hibernate.search.engine.search.projection.dsl.impl.EntityProjectionOptionsStepImpl#toProjection(org.hibernate.search.engine.search.projection.spi.ProjectionMappedTypeContext)` when building a non-loading entity projection.

WDYT?

By the way, if you do this, you can probably remove `org.hibernate.search.engine.search.projection.spi.SearchProjectionBuilderFactory#rootContext` and all related code, as it won't be useful anymore. Maybe in a separate commit so it can be restored later if we introduce `fromRoot`.